soccer/gameplay/skills/capture.py

```diff
@@ -152,5 +132,37 @@ def role_requirements(self):
         reqs.require_kicking = True
         # try to be near the ball
         if main.ball().valid:
-            reqs.destination_shape = main.ball().pos
+            reqs.cost_func = lambda r: reqs.position_cost_multiplier * find_robot_intercept_point(
+                r).dist_to(r.pos)
+
         return reqs
+
+
+def find_robot_intercept_point(robot):
```

Comment:
This function is giving me a ton of deja vu, does it exist somewhere else already? I could just be misthinking though.